dtfs-central-api/src/v1/controllers/portal/facility/update-facility.controller.js

```diff
@@ -12,51 +13,56 @@ const withoutId = (obj) => {
   return cleanedObject;
 };
 
-const updateFacility = async (facilityId, facilityBody, dealId, user, routePath) => {
-  if (ObjectId.isValid(dealId) && ObjectId.isValid(facilityId)) {
-    const collection = await db.getCollection(MONGO_DB_COLLECTIONS.FACILITIES);
+const updateFacility = async ({ facilityId, facilityUpdate, dealId, user, routePath, auditDetails }) => {
+  if (!ObjectId.isValid(dealId) || !ObjectId.isValid(facilityId)) {
+    return { status: 400, message: 'Invalid Deal or Facility Id' };
+  }
 
-    const update = { ...facilityBody, dealId: ObjectId(dealId), updatedAt: Date.now() };
+  const collection = await db.getCollection(MONGO_DB_COLLECTIONS.FACILITIES);
 
-    const findAndUpdateResponse = await collection.findOneAndUpdate({ _id: { $eq: ObjectId(facilityId) } }, $.flatten(withoutId(update)), {
-      returnNewDocument: true,
-      returnDocument: 'after',
-    });
+  const auditRecord = generateAuditDatabaseRecordFromAuditDetails(auditDetails);
 
-    const { value: updatedFacility } = findAndUpdateResponse;
+  const update = { ...facilityUpdate, dealId: ObjectId(dealId), updatedAt: Date.now(), auditRecord };
 
-    if (routePath === PORTAL_ROUTE && user) {
-      // update the deal so that the user that has edited this facility,
-      // is also marked as editing the associated deal
+  const findAndUpdateResponse = await collection.findOneAndUpdate({ _id: { $eq: ObjectId(facilityId) } }, $.flatten(withoutId(update)), {
+    returnNewDocument: true,
+    returnDocument: 'after',
+  });
 
-      await updateDealEditedByPortal(dealId, user);
-    }
+  const { value: updatedFacility } = findAndUpdateResponse;
+
+  if (routePath === PORTAL_ROUTE && user) {
+    // update the deal so that the user that has edited this facility,
+    // is also marked as editing the associated deal
 
-    return updatedFacility;
+    await updateDealEditedByPortal(dealId, user);
```

Comment:
I haven't traced this but does this cause an update to the deal or facility as well which needs an audit record?

Reply:
Spoilers! Theres a PR for this here once this one is merged https://github.com/UK-Export-Finance/dtfs2/pull/3252/files (note its currently sat on this one -- so its larger than it should be)